### PR TITLE
Cloneable test in HmacCore seems questionable

### PR DESCRIPTION
**Pull request by WilburZjh**
This is a backport PR from JDK17 PR https://github.com/ibmruntimes/openj9-openjdk-jdk17/commit/b88cc7219e1bd53bda2adefa8cfffb3bfbc0f8f9 to JDK11.

### CODE REVIEW

**Comment by WilburZjh** [comment]
@keithc-ca please help to review.

File: test/jdk/com/sun/crypto/provider/Mac/DigestCloneabilityTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (keithc-ca) [inline test/jdk/com/sun/crypto/provider/Mac/DigestCloneabilityTest.java:2]: This file, including the copyright date and the final newline, should be identical to upstream.

File: test/jdk/java/security/MessageDigest/TestCloneable.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (keithc-ca) [inline test/jdk/java/security/MessageDigest/TestCloneable.java:2]: This file, including the copyright date and the final newline, should be identical to jdk17.

File: test/jdk/java/security/Signature/TestCloneable.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (keithc-ca) [inline test/jdk/java/security/Signature/TestCloneable.java:2]: This file, including the copyright date and the final newline, should be identical to upstream.

File: src/java.base/share/classes/com/sun/crypto/provider/HmacCore.java [hunk 4]
```diff
@@ -22,6 +22,11 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2023, 2023 All Rights Reserved
+ * ===========================================================================
+ */
```
Comment (keithc-ca) [inline src/java.base/share/classes/com/sun/crypto/provider/HmacCore.java:29]: This is only appropriate if some of the content is contributed by IBM: please remove these lines.

File: src/java.base/share/classes/java/security/MessageDigest.java [hunk 5]
```diff
@@ -22,6 +22,11 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2023, 2023 All Rights Reserved
+ * ===========================================================================
+ */
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/security/MessageDigest.java:29]: Please remove these lines.

File: src/java.base/share/classes/sun/security/provider/DigestBase.java [hunk 6]
```diff
@@ -22,6 +22,11 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2023, 2023 All Rights Reserved
+ * ===========================================================================
+ */
```
Comment (keithc-ca) [inline src/java.base/share/classes/sun/security/provider/DigestBase.java:29]: Please remove these lines.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/sun/security/provider/DigestBase.java (hunk 6 above)
Comment (WilburZjh) [reply src/java.base/share/classes/sun/security/provider/DigestBase.java]: Removed

File: src/java.base/share/classes/java/security/Signature.java [hunk 7]
```diff
@@ -22,6 +22,11 @@
  * or visit www.oracle.com if you need additional information or have any
  * questions.
  */
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2023, 2023 All Rights Reserved
+ * ===========================================================================
+ */
```
[the remark this answers is not in the document]
Comment (WilburZjh) [reply src/java.base/share/classes/java/security/Signature.java]: Removed

File: src/java.base/share/classes/java/security/MessageDigest.java (hunk 5 above)
Comment (WilburZjh) [reply src/java.base/share/classes/java/security/MessageDigest.java]: Removed

File: src/java.base/share/classes/com/sun/crypto/provider/HmacCore.java (hunk 4 above)
Comment (WilburZjh) [reply src/java.base/share/classes/com/sun/crypto/provider/HmacCore.java]: Removed

File: test/jdk/java/security/Signature/TestCloneable.java (hunk 3 above)
Comment (WilburZjh) [reply test/jdk/java/security/Signature/TestCloneable.java]: Updated

File: test/jdk/java/security/MessageDigest/TestCloneable.java (hunk 2 above)
Comment (WilburZjh) [reply test/jdk/java/security/MessageDigest/TestCloneable.java]: Updated

File: test/jdk/com/sun/crypto/provider/Mac/DigestCloneabilityTest.java (hunk 1 above)
Comment (WilburZjh) [reply test/jdk/com/sun/crypto/provider/Mac/DigestCloneabilityTest.java]: Updated

File: test/jdk/java/security/MessageDigest/TestCloneable.java (hunk 2 above)
Comment (keithc-ca) [reply test/jdk/java/security/MessageDigest/TestCloneable.java]: The copyright date should be 2020; see https://github.com/ibmruntimes/openj9-openjdk-jdk17/blob/openj9/test/jdk/java/security/Signature/TestCloneable.java#L2.

**Comment by keithc-ca** [comment]
In the future, please ensure commit messages contain enough information that someone reviewing it years down the road will understand the change and why it was made.

**Comment by keithc-ca** [comment]
Jenkins test sanity osx jdk11

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
Does this update need to go into the the 0.40 release?

**Comment by keithc-ca** [comment]
> Does this update need to go into the the 0.40 release?

I think it should (assuming it doesn't cause any issues in the main branch).

**Comment by pshipton** [comment]
Created https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/668 for 0.40

**Pull request merged**